### PR TITLE
SCRUM-4657: add references to Variant FMS load

### PR DESCRIPTION
**Pull request by abecerra**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/VariantFmsDTOValidator.java [hunk 1]
```diff
@@ -176,7 +167,29 @@ public Long validateVariant(VariantFmsDTO dto, List<Long> idsAdded, BackendBulkD
...
+			}
+			if (CollectionUtils.isNotEmpty(validatedReferences)) {
+				if (variant.getReferences() == null) {
+					variant.setReferences(new ArrayList<>());
+				} else {
+					variant.getReferences().clear();
+				}
+				variant.getReferences().addAll(validatedReferences);
+			}
+		}
```
Comment (markquintontulloch) [inline src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/VariantFmsDTOValidator.java:193]: What happens if `dto.getReferences()` is empty?  You're not clearing any existing references from the database.  There is code in the AlleleDTOValidator that can basically be copied and pasted here.

File: src/test/java/org/alliancegenome/curation_api/VariantFmsITCase.java [hunk 2]
```diff
@@ -111,10 +111,10 @@ public void variantFmsBulkUpload() throws Exception {
...
-			body("entity.crossReferences[0].resourceDescriptorPage.name", is("homepage"));
-
+			body("entity.crossReferences[0].resourceDescriptorPage.name", is("homepage")).
+			body("entity.references[0].crossReferences[0].referencedCurie", is(pubmedId));
 	}
-	
+
 	@Test
 	@Order(2)
 	public void variantFmsBulkUploadUpdate() throws Exception {
```
Comment (markquintontulloch) [inline src/test/java/org/alliancegenome/curation_api/VariantFmsITCase.java:120]: Need to check that update of references works as expected in this test

File: src/test/java/org/alliancegenome/curation_api/VariantFmsITCase.java [hunk 3]
```diff
@@ -196,7 +196,7 @@ public void variantFmsBulkUploadEmptyRequiredFields() throws Exception {
 		params.put("Locations", createCountParams(1, 0, 1, 0));
 		checkBulkLoadRecordCounts(variantFmsBulkPostEndpoint, variantFmsTestFilePath + "ER_05_empty_allele_id.json", params);
 	}
-	
+
 	@Test
 	@Order(5)
 	public void variantFmsBulkUploadInvalidFields() throws Exception {
```
Comment (markquintontulloch) [inline src/test/java/org/alliancegenome/curation_api/VariantFmsITCase.java:202]: Should have a test for invalid reference here

**Review by markquintontulloch — CHANGES_REQUESTED** [review]
Should really also have integration tests for updates with missing/empty references list.  These would have caught the bug in the validation code. 

As adding the column to the UI, also need to add a UI test to the VariantsTable.test

File: src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/VariantFmsDTOValidator.java (hunk 1 above)
Comment (abecerra) [reply src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/VariantFmsDTOValidator.java:193]: Changed as AlleleDTOValidator

**Comment by markquintontulloch** [comment]
> Should really also have integration tests for updates with missing/empty references list. These would have caught the bug in the validation code.
> 
> As adding the column to the UI, also need to add a UI test to the VariantsTable.test

@abecerra These comments still need to be addressed

**Review by markquintontulloch — APPROVED** [review]
